### PR TITLE
Add management support for the multicluster-observability-operator

### PR DESCRIPTION
**Pull request by periklis**
The following PR prepares the addon to be managed by the [multicluster-observability-operator](stolostron/multicluster-observability-operator/). Two major changes are implemented:
- Support the proposed platform/user-workloads observability capabilities as key-values in the `AddonDeploymentConfig`.
- Use as a default deployment namespace `open-cluster-management-observability`

Furthermore the proposed implementation adds validation support in the logging handler for:
- Validate that at least one `ClusterLogForwarder` pipeline includes `infrastructure`  and/or `audit` input reference when `platformLogsCollection` set.
- Validate that at least one `ClusterLogForwarder` pipeline includes `application` input reference when `userWorkloadLogsCollection` set.

The above implementations are not comprehensive but illustrative on how the capabilities defined in the `AddonDeploymentConfig` can be used within the addon.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (iblancasa) [inline CONTRIBUTING.md:122]: This should not be together. The Instrumentation CR injects libraries that provide metrics + traces and, in the future, I think they will include logs.

**Review by iblancasa — CHANGES_REQUESTED** [review]
Before merging this PR, we need to address some of the comments/questions/concerns that were made in the proposal.

Also, if I understood well, there is a meeting soon that can change the proposal.

File: CONTRIBUTING.md
Comment (pavolloffay) [inline CONTRIBUTING.md:89]: What are the semantics of the value? 

In the proposal https://hackmd.io/aBUzPTEZRuCPp_kZqm4x2A?view it was agreed to support multiple stacks for a single capability.

Comment (pavolloffay) [reply CONTRIBUTING.md:89]: I would prefer to document it here, that the value can be a list

**Review by pavolloffay — CHANGES_REQUESTED** [review]

**Review by pavolloffay — CHANGES_REQUESTED** [review]
Thanks for putting this together and moving from the proposal. I would like to clarify the `AddOnDeploymentConfig` and how multiple options of the same capability are supported.

File: CONTRIBUTING.md
Comment (periklis) [reply CONTRIBUTING.md:89]: @pavolloffay My intention was to add on the code base what we will start supporting now and move the hack.md and all the agreements made there into a formal ADR document. Is that ok for you?

File: CONTRIBUTING.md
Comment (periklis) [reply CONTRIBUTING.md:122]: Good point, I can expand the markdown into separate sections then.

File: CONTRIBUTING.md
Comment (periklis) [reply CONTRIBUTING.md:89]: > What are the semantics of the value?

I picket the FQDN version over the simplified version to keep the path for version migrations open. Some of our supported APIs are declared as alpha, other as beta while other as stable. My intention is here for leave room for individual contributors to move at their own pace. AFAIU opentelemetry APIs are moving currently at a faster and innovative pace than others.

**Review by JoaoBraveCoding — COMMENTED** [review]
Overall lgtm

File: CONTRIBUTING.md
Comment (periklis) [reply CONTRIBUTING.md:111]: Can you fix this in #55 please. Please add a more comprehensive list of supported targets.

File: CONTRIBUTING.md
Comment (periklis) [reply CONTRIBUTING.md:122]: I am going to remove this `& Instrumentation` until you guys contribute reconciling this resource in the addon.

**Comment by periklis** [comment]
> Thanks for putting this together and moving from the proposal. I would like to clarify the `AddOnDeploymentConfig` and how multiple options of the same capability are supported.

The use of multiple values in an `AddOnDeploymentConfig` will look like this, e.g.:
```yaml
customizedVariables:
  - name: userWorkloadLogsCollection
    value: clusterlogforwarders.v1.logging.openshift.io;opentelemetrycollectors.v1beta1.opentelemetry.io
```

translates in the `options.go` handling to:

```go
case KeyUserWorkloadLogsCollection:
  if keyvalue.Value == string(ClusterLogForwarderV1) {
	  opts.UserWorkloads.Enabled = true
	  opts.UserWorkloads.Logs.CollectionEnabled = true
	  opts.UserWorkloads.Logs.Kinds = append(opts.UserWorkloads.Logs.Kinds, ClusterLogForwarderV1)
  }
  if keyvalue.Value == string(OpenTelemetryCollectorV1beta1) {
	  opts.UserWorkloads.Enabled = true
	  opts.UserWorkloads.Logs.CollectionEnabled = true
	  opts.UserWorkloads.Logs.Kinds = append(opts.UserWorkloads.Logs.Kinds, OpenTelemetryCollectorV1beta1)
  }					
```

__Note__: This is just pseudo-code and not meant to be implemented in this PR yet. I am leaving the room open for you guys to add support for multiple values when you are ready with each signal. WDYT?

File: CONTRIBUTING.md
Comment (JoaoBraveCoding) [reply CONTRIBUTING.md:111]: Done in https://github.com/rhobs/multicluster-observability-addon/pull/55/commits/dfddda3a7cef7c31b28a94d210e21aefc38dca50

File: CONTRIBUTING.md
Comment (pavolloffay) [inline CONTRIBUTING.md:96]: could we document the semantical meaning of the value? Especially that it can be a list?

**Review by pavolloffay — COMMENTED** [review]
LGTM (after we clarified

File: CONTRIBUTING.md
Comment (pavolloffay) [inline CONTRIBUTING.md:114]: nit: we are on the new version opentelemetrycollectors.v1beta1.opentelemetry.io

File: CONTRIBUTING.md
Comment (pavolloffay) [inline CONTRIBUTING.md:91]: nit: we are on the next version opentelemetrycollectors.v1beta1.opentelemetry.io

**Review by pavolloffay — APPROVED** [review]

File: demo/addon-config/charts/logging/templates/clusterlogforwarder.yaml
Comment (iblancasa) [inline demo/addon-config/charts/logging/templates/clusterlogforwarder.yaml:5]: I though we were not maintaining the `demo` anymore

**Review by iblancasa — CHANGES_REQUESTED** [review]

**Review by iblancasa — APPROVED** [review]

File: CONTRIBUTING.md
Comment (periklis) [reply CONTRIBUTING.md:114]: yes that's a small miss when editing the md file.

File: demo/addon-config/charts/logging/templates/clusterlogforwarder.yaml
Comment (iblancasa) [reply demo/addon-config/charts/logging/templates/clusterlogforwarder.yaml:5]: Was a nit. I clicked request changes by mistake.

And I don't know why this file was selected. I meant the files from `demo`.

Comment (periklis) [reply demo/addon-config/charts/logging/templates/clusterlogforwarder.yaml:5]: I agree we need to remove them at some point in time but I suggest to do this separately.

**Pull request merged**
